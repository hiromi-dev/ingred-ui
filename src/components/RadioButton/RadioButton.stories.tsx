import { action } from "@storybook/addon-actions";

import React from "react";
import RadioButton, { RadioButtonSize } from "./";

export default {
  title: "RadioButton",
  parameters: {
    components: RadioButton
  }
};

export const Overview = () => (
  <>
    <div style={{ marginBottom: 16 }}>
      <RadioButton
        defaultChecked={true}
        name="group"
        onChange={action("onChange1")}
      >
        ラジオボタン
      </RadioButton>
    </div>
    <div>
      <RadioButton name="group" onChange={action("onChange2")}>
        ラジオボタン
      </RadioButton>
    </div>
  </>
);

export const WithControled = () => (
  <RadioButton checked name="group" onChange={action("onChange")} />
);

export const WithLongLabel = () => (
  <>
    <div style={{ marginBottom: 16 }}>
      <RadioButton
        name="group"
        size={RadioButtonSize.MEDIUM}
        onChange={action("onChange")}
      >
        {"長いラベル ".repeat(15).trim()}
      </RadioButton>
    </div>
  </>
);
