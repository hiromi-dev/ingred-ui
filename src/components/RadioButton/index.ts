export { default, RadioButtonSize } from "./RadioButton";
