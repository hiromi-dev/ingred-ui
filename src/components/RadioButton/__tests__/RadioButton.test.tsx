import * as React from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "@testing-library/react";
import RadioButton from "../RadioButton";
import { renderWithThemeProvider } from "../../../utils/renderWithThemeProvider";

describe("RadioButton component testing", () => {
  afterEach(cleanup);

  test("RadioButton", () => {
    const { asFragment } = renderWithThemeProvider(
      <RadioButton>ラジオボタン</RadioButton>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
