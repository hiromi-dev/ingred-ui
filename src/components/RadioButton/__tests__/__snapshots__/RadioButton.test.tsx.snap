// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton component testing RadioButton 1`] = `
<DocumentFragment>
  <label
    class="sc-AxjAm fwXpkS"
  >
    <input
      type="radio"
    />
    <div
      class="sc-AxirZ fFxNUI"
    />
    <span
      class="sc-AxiKw dTGuxX"
    >
      ラジオボタン
    </span>
  </label>
</DocumentFragment>
`;
